### PR TITLE
[MM-49024] Remove v0.7 app manifest workaround

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Compatibility with the app manifest of v0.7 is no longer needed. All apps have been updated to the new schema.

This PR removes the workaround that supported v0.7 manifests.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-49024

### CODE REVIEW

File: apps/manifest.go [hunk 1]
```diff
@@ -146,29 +139,16 @@ type Manifest struct {
 // Thus, custom functions to encode/decode JSON, with backwards compatibility
 // support for App and Manifest.
 func DecodeCompatibleManifest(data []byte) (m *Manifest, err error) {
```
Comment (hanzei) [inline apps/manifest.go:141]: I would like to rename this function (and the app one) to  `UnmarshalManifest` (and `UnmarshalApp`) as they are not longer about compatibility. Thoughts?

**Review by levb — APPROVED** [review]
👍 Thank you!

File: apps/manifest.go (hunk 1 above)
Comment (hanzei) [reply apps/manifest.go:141]: @levb What do you think?

Comment (levb) [reply apps/manifest.go:141]: I thought about it too, as I was reviewing the code. 0/5 `Parse[Manifest|App]JSON`? But I don't think it's a must do.

**Pull request merged**